src/dailyai/pipeline/pipeline.py

```diff
@@ -68,21 +70,35 @@ async def run_pipeline(self):
 
         This method will exit when an EndStreamQueueFrame is placed on the sink queue.
         No more frames will be placed on the sink queue after an EndStreamQueueFrame, even
-        if it's not the last frame yielded by the last frame_processor in the pipeline.."""
+        if it's not the last frame yielded by the last frame_processor in the pipeline..
+        """
 
         if self.source is None or self.sink is None:
             raise ValueError("Source or sink queue not set")
 
         try:
             while True:
                 initial_frame = await self.source.get()
-                async for frame in self.run_pipeline_recursively(initial_frame, self.processors):
+                async for frame in self.run_pipeline_recursively(
+                    initial_frame, self.processors
+                ):
                     await self.sink.put(frame)
 
-                if isinstance(initial_frame, EndFrame) or isinstance(initial_frame, EndPipeFrame):
+                if isinstance(initial_frame, EndFrame) or isinstance(
+                    initial_frame, EndPipeFrame
+                ):
                     break
         except asyncio.CancelledError:
             # this means there's been an interruption, do any cleanup necessary here.
             for processor in self.processors:
                 await processor.interrupted()
             pass
+
+    async def queue_frames(self, frames: List[Frame]):
+        """Insert frames directly into a pipeline. This is typically used inside a transport
+        participant_joined callback to prompt a bot to start a conversation, for example.
+        """
+        if not isinstance(frames, List):
+            frames = [frames]
+        for f in frames:
+            await self.source.put(f)
```

Comment:
If `frames` can be a list or an individual frame, probably best to note that in the type definition too. (`frames: Frame | List[Frame]`)